Dockerfile

```diff
@@ -21,7 +21,7 @@ RUN apt-get update \
   && rm -rf /var/lib/apt/lists/* \
   && pip install -U pip \
   && pip install -r /tmp/RFQuack/requirements.pip \
-  && platformio platform install espressif32
+  && pio pkg install -g --platform "platformio/espressif32@^5.1.0"
```

Comment:
@ResistanceIsUseless just curious, as I'm not familiar with CI/CD best practices. Is it correct to pin a version of dependencies?

Reply:
It's definitely good practice for stability. Generally though in enterprise environments you'll have a dependency scanner telling you which ones are vulnerable and you need to upgrade. We could probably do that automatically with github dependabot and have it update.

I only updated all the packages in requirements.pip because platformio required it to get `platformio platform install espressif32` working and move to the new format of `pio pkg install -g --platform "platformio/espressif32@^5.1.0"` we could probably remove `^5.1.0"` and be fine though. I forget why I picked that version. 

In the end it's a calculation of risk/stability/time to manage.